### PR TITLE
Adds support for n-choose-k type permutations, no replacement

### PR DESCRIPTION
**Pull request by SJCaldwell**
Saw there was a permutations algorithm, but nothing that supported combinations. Never written flow before, but I think I got this gist?

I'd be happy to write some more combinatorics algs if you guys are interested!

### CODE REVIEW

File: src/combinations.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* @flow */
+'use strict';
+/**
+ * Implementation of Combinations
```
Comment (tmcw) [inline src/combinations.js]: Could this give a more informative summary of the functionality, like the [existing functions](http://simplestatistics.org/docs/)? While it's good to link to Wikipedia we want simple-statistics to be informative without needing to follow links. Something like:

> Combinations are unique subsets of a collection - in this case, `k` elements from a collection at a time.

File: src/combinations.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * https://en.wikipedia.org/wiki/Combination
+ * @param {Array} elements any type of data
+ * @param {int} k the number of objects in each group (without replacement)
+ * @returns {Array<Array>} array of permutations
```
Comment (tmcw) [inline src/combinations.js]: Mind adding an example with `@example` here?

File: test/combinations.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/* eslint no-shadow: 0 */
+'use strict';
+
+var test = require('tap').test;
+var ss = require('../');
+
+test('stringHeap', function(t) {
```
Comment (tmcw) [inline test/combinations.test.js]: Should this be

```js
test('combinations', function(t) {
```

**Comment by SJCaldwell** [comment]
Thanks for all the feedback @tmcw, I'll make these changes and push again!

**Comment by SJCaldwell** [comment]
Looking at the new checks, seems I'm still failing.
@tmcw, is this something to do with flow? The code runs but this appears to have more to do with styling.

Sorry if this is super obvious, I've never worked in flow before :P
```
/home/ubuntu/simple-statistics/src/combinations.js
  14:66  error  Missing space before opening brace             space-before-blocks
  15:3   error  Expected indentation of 4 spaces but found 2   indent
  16:3   error  Expected indentation of 4 spaces but found 2   indent
  17:3   error  Expected indentation of 4 spaces but found 2   indent
  18:3   error  Expected indentation of 4 spaces but found 2   indent
  19:3   error  Expected indentation of 4 spaces but found 2   indent
  21:3   error  Expected indentation of 4 spaces but found 2   indent
  21:40  error  Missing space before opening brace             space-before-blocks
  22:5   error  Expected indentation of 6 spaces but found 4   indent
  22:17  error  Missing space before opening brace             space-before-blocks
  23:7   error  Expected indentation of 8 spaces but found 6   indent
  23:28  error  There should be no space after '['             array-bracket-spacing
  23:42  error  There should be no space before ']'            array-bracket-spacing
  25:7   error  Expected indentation of 8 spaces but found 6   indent
  25:57  error  Infix operators must be spaced                 space-infix-ops
  25:80  error  Infix operators must be spaced                 space-infix-ops
  26:7   error  Expected indentation of 8 spaces but found 6   indent
  26:63  error  Missing space before opening brace             space-before-blocks
  27:9   error  Expected indentation of 10 spaces but found 8  indent
  28:9   error  Expected indentation of 10 spaces but found 8  indent
  29:9   error  Expected indentation of 10 spaces but found 8  indent
  33:3   error  Expected indentation of 4 spaces but found 2   indent

/home/ubuntu/simple-statistics/test/combinations.test.js
  14:15  error  A space is required after ','  comma-spacing
  15:15  error  A space is required after ','  comma-spacing
  16:15  error  A space is required after ','  comma-spacing
```

**Comment by tmcw** [comment]
^ Those errors are part of our [eslint](http://eslint.org/) ruleset - making sure that the different algorithms in the module all have the same code style, like indentation, spacing, and so on.

**Comment by tmcw** [comment]
👍 Great, looks like all this needs is to be included in `index.js`, probably around here: https://github.com/simple-statistics/simple-statistics/blob/master/index.js#L45

```js
ss.combinations = require('./src/combinations');
```

**Comment by SJCaldwell** [comment]
Out of curiosity, is it possible to have eslint automatically format code, or do you just fix changes manually each time?

File: src/combinations.js (hunk 1 above)
Comment (SJCaldwell) [inline src/combinations.js]: I think I fixed this? I kind of just used your suggestion :P

**Comment by tmcw** [comment]
Yep, you can run `./node_modules/.bin/eslint --fix` in some cases - the [rules](http://eslint.org/docs/rules/) have wrench icons for ones that can be auto-fixed.

**Pull request merged**